protocol/messenger.go

```diff
@@ -1216,11 +1223,10 @@ func (m *Messenger) CreateGroupChatWithMembers(ctx context.Context, name string,
 	m.allChats[chat.ID] = &chat
 
 	_, err = m.dispatchMessage(ctx, common.RawMessage{
```

Comment:
I'd suggest to wrap similar calls to function like `dispatchMembershipUpdateMessage` to avoid simultaneous changes later